### DISCUSSION TITLE
TipTap / ProseMirror import issue

### DISCUSSION THREAD

**Discussion by SamDuvall** [ueberdosis/tiptap#3282] [General Questions]
I'm having issues importing modules from both Tiptap _and_ ProseMirror.

- Importing the `prosemirror-*` modules causes the [ProseMirror import modules](https://github.com/ProseMirror/prosemirror-model/blob/master/package.json#L10) to be imported.
- Importing the `@tiptap/*` modules causes the [Tiptap CommonJS modules](https://github.com/ueberdosis/tiptap/blob/main/packages/core/package.json#L18) to be required, which in turn require the [ProseMirror CommonJS modules](https://github.com/ProseMirror/prosemirror-model/blob/master/package.json#L11).

Thus, I have two different versions of ProseMirror running through my application, which causes a variety of different problems. I have a larger project that has hundreds of `requires`, so it's not an easy move to convert the whole project over the ESM. For the short-term, I was able to alias the `@tiptap/*` modules to their .esm.js files in webpack, but that seems like a hack. Is there a better way I can solve this issue without having to convert my whole project to ESM?

Thanks!

**Comment by kaitnoone** [comment]
Hey @SamDuvall, did you ever find any resolution here?

**Comment by SamDuvall** [comment]
Hi @kaitnoone, I believe that TipTap has since made a change to how they build & export their repos AND all of their reports are now pointing to the same `prosemirror-*` libraries. Are you having a similar issue?

**Comment by Bunkerbewohner** [comment]
Hi! There have been some changes recently in how the prosemirror dependencies are handled. Previously it was required to add prosemirror as peer dependencies yourself. Now there is a new @tiptap/pm package which pins the correct versions.

See https://tiptap.dev/blog/new-pm-package-and-upgrade-guide-for-beta-210 for instructions.

**Reply by kaitnoone** [reply]
Hey @Bunkerbewohner thanks for the reply! I've run through that install guide a few times and still running into issues resolving dependencies. I've been looking at this far too long so figured I'd post to see if I'm missing something obvious...

```

These dependencies were not found:
* @tiptap/pm/commands in ./node_modules/@tiptap/core/dist/index.js
* @tiptap/pm/keymap in ./node_modules/@tiptap/core/dist/index.js
* @tiptap/pm/model in /node_modules/@tiptap/core/dist/index.js
* @tiptap/pm/schema-list in /node_modules/@tiptap/core/dist/index.js
* @tiptap/pm/state in /node_modules/@tiptap/core/dist/index.js,./node_modules/@tiptap/extension-bubble-menu/dist/index.jsand1 other
* @tiptap/pm/transform in /node_modules/@tiptap/core/dist/index.js
* @tiptap/pm/view in ./node_modules/@tiptap/core/dist/index.js
To install them, you can run: pm install --save @tiptap/pm/commands @tiptap/pm/keymap @tiptap/pm/model @tiptap/pm/schema-list @tiptap/pm/state @tiptap/pm/transform @tiptap/pm/view`

```
